hack/verify-bash-completion.sh

```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+STI_ROOT=$(dirname "${BASH_SOURCE}")/..
+source "${STI_ROOT}/hack/common.sh"
+
+pushd "${STI_ROOT}" >/dev/null
+
+mv contrib/bash/s2i contrib/bash/s2i-proposed
+hack/update-generated-completions.sh
+
+ret=0
+diff -Naupr contrib/bash/s2i contrib/bash/s2i-proposed || ret=$?
+
+mv contrib/bash/s2i-proposed contrib/bash/s2i
+if [[ $ret -eq 0 ]]
+then
+  echo "SUCCESS: Generated completions up to date."
+else
+  echo "FAILURE: Generated completions out of date. Please run hack/update-generated-completions.sh"
+  exit 1
```

Comment:
Can we do it in a subshell? IMHO it's bad practice to change the current working directory when you run some script.
Apart from `cd` itself, a developer don't expect that running a command/script will change his working directory.